### PR TITLE
add check for .DS_Store files

### PR DESCRIPTION
**Pull request by maelle**
@mpadge I'm not too sure how I should approach adding a check, but I'd like to learn. :smiley_cat:

### CODE REVIEW

**Comment by mpadge** [comment]
Thanks! I found a really good definition of "scrap" somewhere, and then promptly lost it again. It has `.DS_Store`, but also just a couple more. I'll wait until i find that, amend here accordingly, then and most importantly, extend this into being a full-blown check that brings up a :heavy_multiplication_x: in the `summary()` function both so you can see how it's done, and so that the process can be documented in the [metadocs](https://ropensci-review-tools.readthedocs.io).

**Comment by maelle** [comment]
The definition I can think of is

``` r
gitignore::gi_fetch_templates("R")
#> 
#> # Created by https://www.toptal.com/developers/gitignore/api/r
#> # Edit at https://www.toptal.com/developers/gitignore?templates=r
#> 
#> ### R ###
#> # History files
#> .Rhistory
#> .Rapp.history
#> 
#> # Session Data files
#> .RData
#> 
#> # User-specific files
#> .Ruserdata
#> 
#> # Example code in package build process
#> *-Ex.R
#> 
#> # Output files from R CMD build
#> /*.tar.gz
#> 
#> # Output files from R CMD check
#> /*.Rcheck/
#> 
#> # RStudio files
#> .Rproj.user/
#> 
#> # produced vignettes
#> vignettes/*.html
#> vignettes/*.pdf
#> 
#> # OAuth2 token, see https://github.com/hadley/httr/releases/tag/v0.3
#> .httr-oauth
#> 
#> # knitr and R markdown default cache directories
#> *_cache/
#> /cache/
#> 
#> # Temporary files created by R markdown
#> *.utf8.md
#> *.knit.md
#> 
#> # R Environment Variables
#> .Renviron
#> 
#> # pkgdown site
#> docs/
#> 
#> # translation temp files
#> po/*~
#> 
#> ### R.Bookdown Stack ###
#> # R package: bookdown caching files
#> /*_files/
#> 
#> # End of https://www.toptal.com/developers/gitignore/api/r
```

<sup>Created on 2021-09-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by mpadge** [comment]
Is there a good reason why you've got `scrap` as a  function and not just a variable? (I'm keen to hear a positive answer there!)

**Comment by maelle** [comment]
I was thinking it could be used e.g. in the roxygen2 docs (but now, of course, I'd need to define it outside of the function for that).

File: R/misc-checks.R [hunk 1]
```diff
@@ -338,3 +338,23 @@ left_assign <- function (path) {
...
     return (assigns)
 }
+
+#' Check that the package contains no useless files like `.DS_Store`.
+#' @inheritParams pkg_has_no_scrap
+#' @return 'TRUE' if all files generated by 'roxygen2', otherwise 'FALSE'
+#' @noRd
+pkg_has_no_scrap <- function (path) {
+
+    all_contents <- gert::git_ls ()$path
```
Comment (maelle) [inline R/misc-checks.R:348]: smart :ok_hand:

Comment (mpadge) [reply R/misc-checks.R:348]: Yeah, because main usage is actually intended to be local, so most people will actually have these scrap files

**Pull request merged**